main.tf

```
@@ -15,6 +15,7 @@ locals {
       for rtb_id in try(v.vpc_route_table_ids, []) : {
         rtb_id = rtb_id
         cidr   = v.tgw_destination_cidr
+        tgw_id = v.tgw_id
```

If I might make a suggestion:
```suggestion
        tgw_id = !var.create_tgw ? v.tgw_id : null
```
In situations such as a hub VPC where the TGW is being created, `tgw_id` might reasonably be left off the VPC attachments, so this guards against that.

---

Isn't that handled here https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/pull/83/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR117?

---

That will ensure the reference to the TGW is handled correctly for the route, but you'll still get the error below if you set `create_tgw = true` and don't pass in `tgw_id` to the attachment:

```
│ Error: Unsupported attribute
│
│   on .terraform/modules/hub_account_example.tgw/main.tf line 18, in locals:
│   18:         tgw_id = v.tgw_id
│
│ This object does not have an attribute named "tgw_id".
```

---

It is strange I didn't get this error as I tested both with hub VPC and an attachment without a `tgw_id` and in scenario where I just attach a VPC. But I agree this is perfectly valid.